source/03-components/external-link/external-link.es6.ts

```diff
@@ -0,0 +1,52 @@
+import Drupal from 'drupal';
+import once from 'once';
+
+Drupal.behaviors.externalLink = {
+  attach(context, settings) {
+    const { imagePath } = settings.gesso;
+    const exitDisclaimer = settings?.gesso?.externalLinkExitDisclaimer ?? 'Exit this website';
+    const allowedDomains = settings?.gesso?.externalLinkAllowedDomains ?? [];
+    const allowedLinks = settings?.gesso?.externalLinkAllowedLinks ?? [];
+    const externalLinks: NodeListOf<HTMLAnchorElement> = once(
+      'external-link',
+      "a:not([data-not-external-link], [href=''], [href^='#'], [href^='?'], [href^='/'], [href^='.'], [href^='javascript:'], [href^='mailto:'], [href^='tel:'])",
+      context
+    );
+
+    function linkIsExternal(link: HTMLAnchorElement) {
+      let external = true;
+
+      if (
+        link.host === window.location.host
+      ) {
+        external = false;
+      }
+
+      allowedDomains.forEach(domain => {
+        if (
+          link.host === domain ||
+          link.host.endsWith(`.${domain}`)
+        ) {
+          external = false;
+        }
+      });
+
+      if (allowedLinks.includes(link.href.toLowerCase())) {
+        external = false;
```

Comment:
Another way to do this would be to return as soon as something sets external to false, since if once we know the link is not external, there's no reason to do the other checks.

```
if (link.host === window.location.host) {
  return false;
}
// Other checks
return true;
```